<h1 style="background-color:powderblue;"><h1 style = "font-size:300%;"><center><b>Olympics Data Analysis</b></center></h1></h1>

![](https://media-cldnry.s-nbcnews.com/image/upload/rockcms/2022-02/paris-olympics-kb-2x1-220214-c2809e.jpg)

The Olympics is a major international sporting event that takes place every four years, bringing together athletes from around the world to compete in various sports. The modern Olympic Games were inspired by the ancient Olympic Games held in ancient Greece from the 8th century BCE to the 4th century CE.

The modern Olympics were first held in Athens, Greece, in 1896, and have since become the most prestigious and widely watched sporting event globally. The Games consist of two main editions: the Summer Olympics and the Winter Olympics, each held every four years but staggered, so there is an Olympic Games event every two years.

Key aspects of the Olympics include:

Participation: Athletes from different countries compete in the Games, fostering camaraderie, friendship, and understanding among nations.

Sports: A wide range of sports are featured in the Olympics, including athletics, swimming, gymnastics, soccer, basketball, skiing, ice hockey, and many others.

Host Cities: The Games are hosted by different cities around the world, selected through a competitive bidding process overseen by the International Olympic Committee (IOC).

Olympic Symbols: The Olympic Rings (five interlocking rings representing the continents) and the Olympic Flame (symbolizing the continuity between ancient and modern Games) are iconic symbols of the Olympics.

Olympic Values: The Games are guided by core principles, including excellence, friendship, respect, and fair play.

Ceremonies: The opening and closing ceremonies are grand spectacles that showcase the host country's culture and history.

Legacy: Hosting the Olympics often leaves a lasting legacy of improved infrastructure, sports facilities, and urban development in the host city.

The Olympics have faced challenges and controversies throughout their history, such as boycotts, doping scandals, and financial concerns. Nevertheless, they continue to captivate the world and inspire athletes to achieve their best on the global stage. The Olympics remain a celebration of human athleticism, international cooperation, and the pursuit of sporting excellence.

In the year 2020, Olympics had to happen in the Tokyo city of Japan but due to Covid-19 pandemic they got postponed to the year 2021. Here is an exploratory data analysis for the same olympics games held in the city of Tokyo in 2021.

### Let's get started !

<h1 style="background-color:powderblue;"><b><center>Exploratory Data Analysis</center></b></h1>

In [1]:
# importing important libraries
import pandas as pd
import numpy as np # data pre-processing

import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px # data visualization

import warnings
warnings.filterwarnings("ignore") # additional

In [2]:
%%bash
pip install openpyxl

In [3]:
#loading the data
df_athletes = pd.read_excel("../input/2021-olympics-in-tokyo/Athletes.xlsx")
df_coaches = pd.read_excel("../input/2021-olympics-in-tokyo/Coaches.xlsx")
df_entries_gender = pd.read_excel("../input/2021-olympics-in-tokyo/EntriesGender.xlsx")
df_medals = pd.read_excel("../input/2021-olympics-in-tokyo/Medals.xlsx")
df_teams = pd.read_excel("../input/2021-olympics-in-tokyo/Teams.xlsx")

<h1 style="background-color:powderblue;"><center><b>Knowing and pre-processing of Data</b></center></h1>

In [4]:

print(df_athletes.head())  # first 5 rows of our data
print()
print("Number of rows and columns are :", df_athletes.shape) # number of rows and columns respectively
print()
print(df_athletes.info())
print()
print(df_athletes.nunique())
print()
print(df_athletes.duplicated().sum())
print()
print(df_athletes.isnull().sum())

                Name     NOC           Discipline
0    AALERUD Katrine  Norway         Cycling Road
1        ABAD Nestor   Spain  Artistic Gymnastics
2  ABAGNALE Giovanni   Italy               Rowing
3     ABALDE Alberto   Spain           Basketball
4      ABALDE Tamara   Spain           Basketball

Number of rows and columns are : (11085, 3)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11085 entries, 0 to 11084
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Name        11085 non-null  object
 1   NOC         11085 non-null  object
 2   Discipline  11085 non-null  object
dtypes: object(3)
memory usage: 259.9+ KB
None

Name          11062
NOC             206
Discipline       46
dtype: int64

1

Name          0
NOC           0
Discipline    0
dtype: int64


# Note :
* There are **three columns** in the athletes dataset i.e., **Name, NOC, Discipline**.

* **Name** has all **unique entries** of **name** of all the **participants** taking part in the olympics.

* **NOC** and **Discipline** are having **categorical values** of number of country participants come from and Discipline they are participating in, **has more than one entries** for different participant.

* There is **one duplicate row** and **zero null row**.


In [5]:
print(df_entries_gender.head())
print()
print("Number of rows and columns are :", df_entries_gender.shape)
print()
print(df_entries_gender.info())
print()
print(df_entries_gender.nunique())
print()
print(df_entries_gender.duplicated().sum())
print()
print(df_entries_gender.isnull().sum())

            Discipline  Female  Male  Total
0       3x3 Basketball      32    32     64
1              Archery      64    64    128
2  Artistic Gymnastics      98    98    196
3    Artistic Swimming     105     0    105
4            Athletics     969  1072   2041

Number of rows and columns are : (46, 4)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 46 entries, 0 to 45
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Discipline  46 non-null     object
 1   Female      46 non-null     int64 
 2   Male        46 non-null     int64 
 3   Total       46 non-null     int64 
dtypes: int64(3), object(1)
memory usage: 1.6+ KB
None

Discipline    46
Female        38
Male          41
Total         41
dtype: int64

0

Discipline    0
Female        0
Male          0
Total         0
dtype: int64


# Note :
* Above data has **four columns** where **one** has **str** entries whereas other **three** has **integral** entries.

* Data has **zero duplicate** and **zero null** row.

* Data has **46 rows** and **4 columns**.

* The command **nunique** gives us the **unique value column** has.

In [6]:
print(df_medals.head())
print()
print("Number of rows and columns are :", df_medals.shape)
print()
print(df_medals.info())
print()
print(df_medals.nunique())
print()
print(df_medals.duplicated().sum())
print()
print(df_medals.isnull().sum())

   Rank                    Team/NOC  Gold  Silver  Bronze  Total  \
0     1    United States of America    39      41      33    113   
1     2  People's Republic of China    38      32      18     88   
2     3                       Japan    27      14      17     58   
3     4               Great Britain    22      21      22     65   
4     5                         ROC    20      28      23     71   

   Rank by Total  
0              1  
1              2  
2              5  
3              4  
4              3  

Number of rows and columns are : (93, 7)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93 entries, 0 to 92
Data columns (total 7 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Rank           93 non-null     int64 
 1   Team/NOC       93 non-null     object
 2   Gold           93 non-null     int64 
 3   Silver         93 non-null     int64 
 4   Bronze         93 non-null     int64 
 5   Total          93 non-null

# Note :
* Above data **gives** the **rank-wise medal record** of each country.

* Note that **USA is on the top followed by China and Japan**.

* Data has **zero duplicate** and **zero null row** entry.

* Data has **93 rows** and **7 columns**.



In [7]:
print(df_coaches.head())
print()
print("Number of rows and columns are :", df_coaches.shape)
print()
print(df_coaches.info())
print()
print(df_coaches.nunique())
print()
print(df_coaches.duplicated().sum())
print()
print(df_coaches.isnull().sum())

              Name            NOC  Discipline Event
0  ABDELMAGID Wael          Egypt    Football   NaN
1        ABE Junya          Japan  Volleyball   NaN
2    ABE Katsuhiko          Japan  Basketball   NaN
3     ADAMA Cherif  Côte d'Ivoire    Football   NaN
4       AGEBA Yuya          Japan  Volleyball   NaN

Number of rows and columns are : (394, 4)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 394 entries, 0 to 393
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Name        394 non-null    object
 1   NOC         394 non-null    object
 2   Discipline  394 non-null    object
 3   Event       249 non-null    object
dtypes: object(4)
memory usage: 12.4+ KB
None

Name          381
NOC            61
Discipline      9
Event           6
dtype: int64

1

Name            0
NOC             0
Discipline      0
Event         145
dtype: int64


# Note :
* Above data has **394 rows** and **4 columns**.

* It has record of **coaches** from **different countries and discipline**.

* Data has **one duplicate** row and **zero null** row.

* **All** the **columns** have **str** entries.

In [8]:
print(df_teams.head())
print()
print("Number of rows and columns are :", df_teams.shape)
print()
print(df_teams.info())
print()
print(df_teams.nunique())
print()
print(df_teams.duplicated().sum())
print()
print(df_teams.isnull().sum())

      Name      Discipline                         NOC  Event
0  Belgium  3x3 Basketball                     Belgium    Men
1    China  3x3 Basketball  People's Republic of China    Men
2    China  3x3 Basketball  People's Republic of China  Women
3   France  3x3 Basketball                      France  Women
4    Italy  3x3 Basketball                       Italy  Women

Number of rows and columns are : (743, 4)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 743 entries, 0 to 742
Data columns (total 4 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Name        743 non-null    object
 1   Discipline  743 non-null    object
 2   NOC         743 non-null    object
 3   Event       743 non-null    object
dtypes: object(4)
memory usage: 23.3+ KB
None

Name          146
Discipline     20
NOC            84
Event          36
dtype: int64

0

Name          0
Discipline    0
NOC           0
Event         0
dtype: int64


<h1 style="background-color:powderblue;"><center><b>Data Visualization</b></center></h1>

In [9]:
x = df_athletes.NOC.value_counts()

In [10]:
px.bar(x[:20], title="Top 20 countries in terms of number of participants :")

# Note : 

* Above plot shows **top count of participants** coming **from different countries**.

* **USA send maximum number of participants followed by Japan and Australia**.

In [11]:
x = df_coaches.NOC.value_counts()

In [12]:
px.bar(x[:20], title="Top 20 countries in terms of number of coaches :")

# Note :

* Above plot shows **countries** from where **maximum number of coaches** are coming from.

* **Japan** contributes to **maximum number** of coaches **followed by USA and Spain**.

In [13]:
px.bar(df_medals, x="Team/NOC", y="Total", color="Gold", title="Top Countries in terms of number of medals : ")

#  Note :

* Above plot shows **highest ranked countries as per their medal count**.

* **USA** is on the **top** follwed by **China and Japan**.

* Note that **Germany and Australia** are in **top 5 countries participants** are coming from but **not in top 5 countries having maximum medals**.

* Whereas **USA, China, Japan holds top position in both number of participants and medals won**. 

In [14]:
x = df_athletes.Discipline.value_counts()

In [15]:
px.bar(x[:20])

In [16]:
px.bar(df_entries_gender, x="Discipline", y="Total", color="Discipline", title="Total participants in each Discipline : ")

# Note :

* **Atheletics** has **maximum number of participants**.

* **Swimming** has **second highest number of participants**.

* **Cycling** has **minimum number of participants**.

In [17]:
px.choropleth(df_medals, locations="Team/NOC", 
                    locationmode='country names', color="Team/NOC", 
                    hover_name="Total", range_color=[1,100], 
                    color_continuous_scale="blues", 
                    title='Density of Countries in 2020')

# Note :

* Plotly is pretty much interactive library.

* **Move the cursor over any country** to **know number of medals won by that country**.


In [18]:
px.scatter(df_medals, x="Total", y="Rank",size="Total", color="Team/NOC",hover_name="Team/NOC", log_x=True, size_max=60)

# Note :

* This plot shows **Rank vs Total** medal won by each country. 

* **USA** has biggest circle size and has rank one with **maximum Total** score.